builtin/logical/pki/backend.go

```diff
@@ -460,6 +468,25 @@ func (b *backend) invalidate(ctx context.Context, key string) {
 		b.crlBuilder.markConfigDirty()
 	case key == storageIssuerConfig:
 		b.crlBuilder.invalidateCRLBuildTime()
+	case strings.HasPrefix(key, crossRevocationPrefix):
+		split := strings.Split(key, "/")
+
+		if !strings.HasSuffix(key, "/confirmed") {
+			cluster := split[len(split)-2]
+			serial := split[len(split)-1]
+			// Only process confirmations on the perf primary.
+			b.crlBuilder.addCertForRevocationCheck(cluster, serial)
+		} else {
+			if len(split) >= 3 {
+				cluster := split[len(split)-3]
+				serial := split[len(split)-2]
+				if !isNotPerfPrimary {
```

Comment:
I should move comment on line 477 down here later, and add a note in `crl_util.go` that `removalQueue` is empty anywayys.